code/datums/action.dm

```diff
@@ -580,6 +580,25 @@
 	owner.forceMove(box)
 	owner.playsound_local(box, 'sound/misc/box_deploy.ogg', 50, TRUE)
 
+/datum/action/item_action/agent_box/Grant(mob/M)
+	..()
+	if(owner)
+		RegisterSignal(owner, COMSIG_HUMAN_SUICIDE, .proc/suicide_act)
+
+/datum/action/item_action/agent_box/Remove(mob/M)
+	if(owner)
+		UnregisterSignal(owner, COMSIG_HUMAN_SUICIDE)
+	..()
+
+/datum/action/item_action/agent_box/proc/suicide_act(datum/source)
```

Comment:
Every signal handler needs `SIGNAL_HANDLER` on it.

https://github.com/tgstation/tgstation/blob/master/.github/guides/STANDARDS.md#signal-handlers